aws/resource_aws_globalaccelerator_accelerator_test.go

```diff
@@ -95,6 +103,73 @@ func testSweepGlobalAcceleratorAccelerators(region string) error {
 	return sweeperErrs.ErrorOrNil()
 }
 
+func sweepGlobalAcceleratorListeners(conn *globalaccelerator.GlobalAccelerator, acceleratorArn *string) *multierror.Error {
```

Comment:
Nit: This setup is okay given that there is a hard dependency on removing these, however registering the sweepers to each resource allows you to do targeted sweeping `-sweep-run=aws_global_accelerator_listener` in addition to also cleaning up the sub-resources if `aws_global_accelerator_accelerator` has `Dependencies: []string{"aws_global_accelerator_listener"}` added to its sweeper.

Reply:
Ok, I figured that since every `aws_globalaccelerator_endpoint_group` depends on an `aws_globalaccelerator_listener`, which depends on an `aws_globalaccelerator_accelerator`, it would make sense to do the sweep in one place. Is there an advantage to being able to sweep, e.g. all Endpoint Groups, while leaving the Listeners and Accelerators?

Only the `aws_globalaccelerator_accelerator` is a stand-alone resource, so the `aws_globalaccelerator_endpoint_group` sweeper would be something like the following pseudocode:
```
for _, accelerator := range accelerators {
    for _, listener := range accelerator.Listeners {
        for _, endpointGroup := range listener.EndpointGroups {
            Delete(endpointGroup)
        }
    }
}
```

Reply:
> Is there an advantage to being able to sweep, e.g. all Endpoint Groups, while leaving the Listeners and Accelerators?

In addition to the targeting flexibility, this may fall under "you don't need it until you need it", such as if other resources needing to declare that specific resource as a sweeper dependency (for now we'd need to declare the top level resource). My only slight point here is more the general practice of not trying to pile all the logic in one sweeper, same as we try to avoid piling multiple resource logic into a single resource. We (of course) also have excellent anti-examples like `aws_iam_user` you recently updated that painstakingly go through dependencies in the top level sweeper. The pseudocode you presented would be the answer here, but honestly any refactoring effort here is not necessary at the moment.